### PR TITLE
make state syncing compatible with new BLS consensus code; last mile

### PR DESCRIPTION
**Pull request by chaosma**
Fixing state syncing breaking with new BLS consensus code. The last mile problem still need to be fix.

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -211,6 +227,29 @@ func (consensus *Consensus) checkConsensusMessage(message consensus_proto.Messag
 	// check consensus Id
 	if consensusID != consensus.consensusID {
 		utils.GetLogInstance().Warn("Wrong consensus Id", "myConsensusId", consensus.consensusID, "theirConsensusId", consensusID, "consensus", consensus)
+		if consensus.IsLeader {
+			return false
+		}
+		// validator send consensus block to state syncing
+		if val, ok := consensus.blocksReceived[consensusID]; ok {
```
Comment (rlan35) [inline consensus/consensus.go]: I would suggest move this logic to the caller level of checkConsensusMessage, by defining and returning different error types and checking the type on the caller side.

Comment (chaosma) [reply consensus/consensus.go]: Sure

File: consensus/consensus_validator.go [hunk 2]
```diff
@@ -12,6 +12,29 @@ import (
 	"github.com/harmony-one/harmony/internal/utils"
 )
 
+func (consensus *Consensus) sendBFTBlockToStateSyncing(consensusID uint32) {
```
Comment (LeoHChen) [inline consensus/consensus_validator.go]: comment please.

File: consensus/consensus_validator.go [hunk 3]
```diff
@@ -12,6 +12,29 @@ import (
...
+	// validator send consensus block to state syncing
+	if val, ok := consensus.blocksReceived[consensusID]; ok {
```
Comment (LeoHChen) [inline consensus/consensus_validator.go]: golang will usual check !ok and return early.

Comment (chaosma) [reply consensus/consensus_validator.go]: This will also return if !ok

**Pull request merged**